db/version_builder.cc

```diff
@@ -1181,68 +1186,84 @@ class VersionBuilder::Rep {
       }
     }
 
-    // <file metadata, level>
-    std::vector<std::pair<FileMetaData*, int>> files_meta;
-    std::vector<Status> statuses;
-    for (int level = 0; level < num_levels_; level++) {
-      for (auto& file_meta_pair : levels_[level].added_files) {
-        auto* file_meta = file_meta_pair.second;
-        // If the file has been opened before, just skip it.
-        if (!file_meta->table_reader_handle) {
-          files_meta.emplace_back(file_meta, level);
-          statuses.emplace_back(Status::OK());
+    Status ret;
+    if (0 < max_load) {
+      // <file metadata, level>
+      std::vector<std::pair<FileMetaData*, int>> files_meta;
+      std::vector<Status> statuses;
+      for (int level = 0; level < num_levels_; level++) {
+        for (auto& file_meta_pair : levels_[level].added_files) {
+          auto* file_meta = file_meta_pair.second;
+          // If the file has been opened before, just skip it.
+          if (!file_meta->table_reader_handle) {
+            files_meta.emplace_back(file_meta, level);
+            statuses.emplace_back(Status::OK());
```

Comment:
The old code depends upon the vector being properly sized when used later since multiple threads access the vector, though at different indexes.  I would say this is "code safety" and/or "thread safety".